### PR TITLE
Handle connection lost on client side

### PR DESCRIPTION
**Pull request by skrater**

### CODE REVIEW

**Review by alexandrevicenzi — APPROVED** [review]

File: client.go [hunk 1]
```diff
@@ -1,77 +1,109 @@
...
-	// create the response queue (let the amqp server to pick a name for us)
-	q, err := ch.QueueDeclare("", false, false, true, false, nil)
+	defer ch.Close()
+
+	// create the response queue
+	_, err = ch.QueueDeclare(
+		c.responseQueueName, // name
+		false,               // durable
+		true,                // auto-delete
+		false,               // exclusive
```
Comment (gfronza) [inline client.go]: why not an exclusive queue? especially now that we are setting a name to it.

**Review by gfronza — CHANGES_REQUESTED** [review]
Looks very good in general, just one note on the queue declaration.

File: client.go (hunk 1 above)
Comment (skrater) [reply client.go]: makes sense, i will change that

**Comment by gfronza** [comment]
Great! Thanks!

**Pull request merged**